signature-envelope-jws.md

```diff
@@ -0,0 +1,158 @@
+
+Example of payload before BASE64URL encoding.
+
+```jsonc
+{
+  "subject": {
+    "mediaType": "application/vnd.oci.image.manifest.v1+json",
+    "digest": "sha256:73c803930ea3ba1e54bc25c2bdc53edd0284c62ed651fe7b00369da519a3c333",
+    "size": 16724,
+    "annotations": {
+        "io.wabbit-networks.buildId": "123"  // user defined metadata
+    }
+  }
+}
+```
+
+## Protected Headers
+
+The JWS envelope for Notary v2 uses following headers
+
+- Registered headers `alg`, `cty`, and `crit`
+- Public headers `io.cncf.notary.signingTime`, `io.cncf.notary.expiry`, `io.cncf.notary.clientID`
+
+Example
+
+```jsonc
+{
+    "alg": "PS384",
+    "cty": "application/vnd.cncf.notary.v2.jws.v1",
+    "io.cncf.notary.signingTime": "2022-04-06 07:01:20.52Z",
+    "io.cncf.notary.expiry": "2022-10-06 07:01:20.52Z",
+    "io.cncf.notary.clientID": "notation/1.0.0",
+    "crit":["io.cncf.notary.expiry"]
+}
+```
+
+- **[`alg`](https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.1)**(*string*): This REQUIRED header defines which algorithm was used to generate the signature.JWS specification defines `alg` as a required header, that MUST be present and MUST be understood and processed by verifier. This [section](#supported-alg-header-values) lists the Notary v2 allowed subset of `alg` values supported by JWS.
+- **[`cty`](https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.10)**(*string*): The REQUIRED header content-type is used to declare the media type of the secured content (the payload). The supported value is `application/vnd.cncf.notary.payload.v1+json`.
+- **`io.cncf.notary.signingTime`**(*string*): This REQUIRED header specifies the time at which the signature was generated. Its value is a RFC 3339 formatted date time.
+- **`io.cncf.notary.expiry`**(*string*): This OPTIONAL header provides a “best by use” time for the artifact, as defined by the signer. Its value is a RFC 3339 formatted date time.
+- **`io.cncf.notary.clientID`**(*string*): This OPTIONAL header provides The version of a client (e.g. Notation) that produced the signature. E.g. “notation/1.0.0”.
+- **[`crit`](https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.11)**(*array of strings*): This OPTIONAL header lists the headers that implementation MUST understand and process. It MUST only contain headers apart from registered headers (e.g. `alg`, `cty`) in JWS specification, therefore this header is only present when the optional `io.cncf.notary.expiry` header is present in the protected headers collection.
+  If present, the value MUST be `["io.cncf.notary.expiry"]`.
+
+## Unprotected Headers
+
+Notary v2 supports only two unprotected headers: `timestamp` and `x5c`.
+
+```jsonc
+{
+    "timestamp": "<Base64(TimeStampToken)>",
+    "x5c": ["<Base64(DER(leafCert))>", "<Base64(DER(intermediateCACert))>", "<Base64(DER(rootCert))>"]
+}
+```
+
+- **`timestamp`** (*string*): This OPTIONAL property is used to store time stamp token.
```

Comment:
Is `"timestamp"` going to be registered as JWS header parameter? If not, shouldn't it be a collision resistant name like the protected header parameters?